inc/3rd-party/plugins/wp-rest-api.php

```diff
@@ -10,23 +10,37 @@
  * @return array Updated URLs
  */
 function rocket_exclude_wp_rest_api( $uri ) {
+	global $wp_rewrite;
+
 	/**
 	 * By default, don't cache the WP REST API.
 	 *
 	 * @since 2.5.12
 	 *
 	 * @param bool false will force to cache the WP REST API
 	 */
-	$rocket_cache_reject_wp_rest_api = apply_filters( 'rocket_cache_reject_wp_rest_api', true );
+	$reject_wp_rest_api = apply_filters( 'rocket_cache_reject_wp_rest_api', true );
 
-	// Exclude WP REST API.
-	if ( function_exists( 'json_get_url_prefix' ) && $rocket_cache_reject_wp_rest_api ) {
-		$uri[] = '/' . json_get_url_prefix() . '/(.*)';
+	if ( ! $reject_wp_rest_api ) {
+		return $uri;
 	}
 
-	if ( class_exists( 'WP_REST_Controller' ) && $rocket_cache_reject_wp_rest_api ) {
-		$uri[] = '/wp-json/(.*)';
-	}
+	/**
+	 * `(/[^/]+)?` is used instead of `(/.+)?` to match only one level.
```

Comment:
edge case but what if the website is installed in a double (or more) level like `/blog/wordpress`?

Reply:
Not related.
What you mention is handled in `get_rocket_cache_reject_uri()`:
```php
function get_rocket_cache_reject_uri() {
	$uris      = get_rocket_option( 'cache_reject_uri', array() );
	$home_root = rocket_get_home_dirname();

	if ( '' !== $home_root ) {
		// The site is not at the domain root, it's in a folder.
		$home_root_escaped = preg_quote( $home_root, '/' );
		$home_root_len     = strlen( $home_root );
```

`rocket_get_home_dirname()` will return `/blog/wordpress`.
To be sure we speak about the same thing:
- Home: `example.com/blog/wordpress/`,
- Admin: `example.com/blog/wordpress/wp-admin/`.

This one in `rocket_exclude_wp_rest_api()` is used on multisite to match this: `example.com/blog/wordpress/**site-2**/wp-json/`. As far as I know we can't have multiple levels here.

Reply:
right I got mixed up

Reply:
Easy to get lost in all this mess >_>